network/tools/network-policy-enforcement-latency/pkg/client.go

```diff
@@ -0,0 +1,548 @@
+	if !ok {
+		klog.Warningf("Failed to convert work queue item of type %T to string key", item)
+		return
+	}
+
+	// Skip Update pod events for pods that aren't tracked since their creation.
+	// Only new pods should be included as targets.
+	if c.skipUpdateForUntrackedPods(podEvent) {
+		return
+	}
+
+	// Get pod from informer's cache.
+	objList, err := c.podInformer.GetIndexer().ByIndex(utils.NameIndex, podEvent.podName)
+	if err != nil {
+		klog.Warningf("Failed to get pod object from the informer's cache, for pod name %q", podEvent.podName)
+		return
+	}
+
+	if len(objList) < 1 {
+		klog.Warningf("Pod object does not exist in the informer's cache, for pod name %q", podEvent.podName)
+		return
+	}
+
+	pod, ok := objList[0].(*corev1.Pod)
+	if !ok {
+		klog.Warningf("processNextPodCreationEvent() failed to convert obj (%T) to *corev1.Pod", objList[0])
+		return
+	}
+
+	if len(pod.Status.PodIP) > 0 {
+		podName := pod.GetName()
+		podCreationTime := pod.GetCreationTimestamp().Time
+		haveMaxTargets := false
+
+		c.targetPodsMap.lock.Lock()
+		_, targetUsed := c.targetPodsMap.mp[podName]
+		if !targetUsed {
+			c.targetPodsMap.mp[podName] = startTime
+			haveMaxTargets = len(c.targetPodsMap.mp) > c.config.expectedTargets
+		}
+		c.targetPodsMap.lock.Unlock()
+
+		// Process only up to the expected number of pods.
+		if haveMaxTargets {
+			c.stopInformerChan()
+			return
+		}
+
+		// Do the measurements only if it hasn't already been done for this pod.
+		if !targetUsed {
+			target := &targetSpec{
+				ip:        pod.Status.PodIP,
+				name:      podName,
+				namespace: pod.GetNamespace(),
+				startTime: startTime,
+			}
+			go c.recordFirstSuccessfulRequest(target, true)
+
+			podAssignedIPLatency := startTime.Sub(podCreationTime)
+			metrics.PodIPAddressAssignedLatency.Observe(podAssignedIPLatency.Seconds())
+			klog.Infof("Test client got pod %q with assigned IP %q, %v after pod creation", podName, pod.Status.PodIP, podAssignedIPLatency)
+		}
+	}
+}
+
+// recordFirstSuccessfulRequest sends curl requests continuously to the provided
+// target IP and target port from the test client config, at 1 second intervals,
+// until the first successful response. It records the latency between the
+// provided start time in the targetSpec, and the time of the first successful
+// response.
+func (c *TestClient) recordFirstSuccessfulRequest(target *targetSpec, forPodCreation bool) {
+	if target == nil || len(target.ip) == 0 {
+		klog.Infof("Skipping a target for policy enforcement latency (podCreation=%t), because the target does not have an IP address assigned. Target: %v", forPodCreation, target)
+		return
+	}
+
+	request := fmt.Sprintf("curl --connect-timeout 1 %s:%d", target.ip, c.config.targetPort)
+	command := []string{"sh", "-c", "-x", request}
+	klog.Infof("Sending requests %q", request)
+
+	for {
+		select {
+		case <-c.MainStopChan:
+			return
+		default:
+		}
+
+		_, err := utils.RunCommand(command)
+		if err == nil {
```

Comment:
if err != nil {
  continue + some comment why it's expected
}
the code if err is not nil?

Reply:
Done.
Added `err != nil {continue}` and a comment for it.
“CURL command returns errors for requests denied by a policy, or when the IP is unreachable. Everything else is considered a success”